R/deltaccd.R

```diff
@@ -87,6 +87,33 @@ calcCCDSimple = function(ref, emat, method = 'spearman', scale = FALSE) {
   return(ccd)}
 
 
+checkVar = function(emat, groupVec) {
+  
+  varCheck = foreach (groupNow = sort(unique(groupVec)), .combine = rbind) %do% {
+    
+    varMat = apply(emat[, groupVec == groupNow], MARGIN = 1, 
+                   FUN = stats::var, na.rm = TRUE)
+    varDt = data.table::as.data.table(varMat, keep.rownames = 'gene')
+    data.table::setnames(varDt, 'varMat', 'variance')
+    varDt[, group := groupNow]
+    
+    zeroVar = varDt[variance == 0]
+    zeroVar[, variance := NULL]
+    
+    return(zeroVar)}
+  
+  return(varCheck)}
+
+
+checkGenes = function(emat, refCor) {
+  geneNames = rownames(refCor)[rownames(refCor) %in% rownames(emat)]
+  if (length(geneNames) < nrow(refCor)) {
+    missingGenes = setdiff(rownames(refCor), geneNames)
+    stop(paste0('The following gene(s) is/are not in the expression matrix: \n',
+                paste0(missingGenes, collapse = '\n')))
+    } else { return(geneNames) }}
```

Comment:
The `return` statement doesn't need to be inside an `else`, do you see why?

Reply:
Yep, if the `if` condition is met, then the function will throw an error before `geneNames` is returned anyway.